### ISSUE TITLE
Doesn't seem to work

### ISSUE THREAD

**Issue by roman01la** [wakatime/atom-wakatime#9]
I've followed all installation steps, did some code for an hour, but still no data in my dashboard. Also I can't see .waka* files being created in my home directory.

**Comment by alanhamlett** [comment]
What OS are you using?

**Comment by roman01la** [comment]
OS X Yosemite 10.10.2

FYI: Vim plugin works just fine.

**Comment by alanhamlett** [comment]
Can you open dev console and check for error messages?

`View -> Developer -> Toggle Developer Tools`

**Comment by roman01la** [comment]
No errors, `WakaTime v4.0.1 loaded.`

**Comment by roman01la** [comment]
Ok. After trying Vim plugin, Atom plugin is working now. Looks like it won't initialize.

**Comment by alanhamlett** [comment]
Ok, so it's working now? Can you go to https://wakatime.com/api/v1/users/current/stats/last_7_days and see if you have logged time from Atom editor?

**Comment by roman01la** [comment]
Yes, everything is perfectly ok. But, for some reason, Atom plugin started to log time only after I tried Vim plugin. It seems to me, that Atom plugin didn't work because of absence of `.wakatime.*` files in home directory. And so it start working only after I tried Vim plugin, which created these files.

**Comment by asennikov** [comment]
Seems like I'm experiencing the same issue, but installing vim plugin didn't actually fix it.
The `.wakatime.*` files are present, I even have some records in `.wakatime.log`, but there is nothing at dashboard.

Can you help me debug this issue somehow properly?

**Comment by alanhamlett** [comment]
@asennikov Sure:

1. what version of Atom are you using?

2. check your dev console for error messages:
`View -> Developer -> Toggle Developer Tools`

3. check for error messages in your `~/.wakatime.log` file. Looks like you did that already. Were they old errors or recent?

4. check how many offline heartbeats are in your local sqlite database by typing these commands:
```
$ sqlite3 ~/.wakatime.db
SQLite version 3.8.5 2014-08-15 22:37:57
Enter ".help" for usage hints.
sqlite> select count(*) from heartbeat;
0
sqlite> .quit
```

5. does the `cli.py` script exist at `~/.atom/packages/wakatime/lib/wakatime-master/wakatime/cli.py`?

6. Can you save a file in Atom, then check if your `last_plugin` attribute updates to atom-wakatime: https://wakatime.com/api/v1/users/current

**Comment by asennikov** [comment]
@alanhamlett seems like it started working from some point in time... Cause I've only started to walk through you hints, and then checked my account and all the data is already there.

Sorry, for not very accurate info - I'll try it again on another machine in the near future.

**Comment by Chrisedmo** [comment]
I'm also having trouble getting this plug in to work. It worked flawlessly in ST2/3.

It's not logging time at all. 

Here is a sample log:

    {"now": "2015/05/01 15:47:13 +0100", "version": "4.0.8", "plugin": "atom-wakatime/4.0.4", "time": 1430491631.750546, "file": "/Users/namehere/Library/Caches/Cleanup At Startup/Transmit/1DCCD2D8-7F4D-4743-8023-295A47D5C542/index.php", "level": "WARNING", "message": "%s"}
    {"now": "2015/05/01 15:47:39 +0100", "version": "4.0.8", "plugin": "atom-wakatime/4.0.4", "time": 1430491658.566221, "file": "/Users/namehere/Library/Caches/Cleanup At Startup/Transmit/45B61692-513E-4C8D-849C-736ADF51C149/index.php", "level": "WARNING", "message": "%s"}
    {"now": "2015/05/05 10:25:55 +0100", "version": "4.0.8", "plugin": "atom-wakatime/4.0.6", "time": 1430817953.595766, "file": "/Users/namehere/Desktop/LOUVRE_EMAIL_SIGS/Gruntfile.js", "level": "WARNING", "message": "%s"}

Atom: 0.196.0, and the latest Wakatime plugin.

Thanks.

**Comment by alanhamlett** [comment]
Can you check for error messages in your Atom Developer Console?

Those messages are from [wakatime cli](https://github.com/wakatime/wakatime), and not sure what is causing those warning messages. You can try turning on `debug = true` in your `~/.wakatime.cfg` file to see if Atom is sending heartbeats.

I've just updated wakatime cli to include the package namespace and line number for all log messages, but you will have to re-install atom-wakatime to get the new wakatime cli version.

**Comment by Chrisedmo** [comment]
Thanks for getting back. I have uninstalled the plugin in and re-installed it. When i did this and ran it, in the dev console, there was an error message about a file not being found, but then it began to download/update the wakatime CLI. So i restarted again, and there was no more errors in the console.

I then set debug to true, and just tested a few files, and this is the output in my log:

    {"now": "2015/05/05 23:14:30 +0100", "package": "logging", "lineno": 1711, "version": "4.0.8", "plugin": "atom-wakatime/4.0.6", "time": 1430864070.577819, "file": "/Users/username/Documents/Sites/rh/.htaccess", "level": "WARNING", "message": "%s"}
    {"now": "2015/05/05 23:14:31 +0100", "package": "wakatime", "lineno": 378, "version": "4.0.8", "plugin": "atom-wakatime/4.0.6", "time": 1430864070.577819, "file": "/Users/username/Documents/Sites/rh/.htaccess", "level": "DEBUG", "message": {"response_code": 201}}

The only thing that is unusual about my Atom install is that I have synced my settings folder, like this:

    $ mv ~/.atom ~/Dropbox/Apps/Atom

    $ ln -s ~/Dropbox/Apps/Atom ~/.atom

**Comment by alanhamlett** [comment]
The `"response_code": 201` means your heartbeat was sent correctly from Atom to the WakaTime api. It should be logging time now, right?

Using dropbox for your .atom folder shouldn't cause any problems.

**Comment by alanhamlett** [comment]
OK, found the issue was caused by not sending heartbeats after saving a file in Atom. It's fixed in v4.0.7.

For previous days, you can uncheck the box in settings which hides projects without writes:
https://wakatime.com/settings/preferences
